packages/backend/src/open_payments/payment/incoming/routes.test.ts

```diff
@@ -82,31 +82,33 @@ describe('Incoming Payment Routes', (): void => {
           externalRef
         }),
       get: (ctx) => incomingPaymentRoutes.get(ctx),
-      getBody: (incomingPayment, list) => ({
-        id: incomingPayment.url,
-        paymentPointer: paymentPointer.url,
-        completed: false,
-        incomingAmount:
-          incomingPayment.incomingAmount &&
-          serializeAmount(incomingPayment.incomingAmount),
-        description: incomingPayment.description,
-        expiresAt: incomingPayment.expiresAt.toISOString(),
-        createdAt: incomingPayment.createdAt.toISOString(),
-        updatedAt: incomingPayment.updatedAt.toISOString(),
-        receivedAmount: serializeAmount(incomingPayment.receivedAmount),
-        externalRef: '#123',
-        ilpStreamConnection: list
-          ? `${config.openPaymentsUrl}/connections/${incomingPayment.connectionId}`
-          : {
-              id: `${config.openPaymentsUrl}/connections/${incomingPayment.connectionId}`,
-              ilpAddress: expect.stringMatching(
-                /^test\.rafiki\.[a-zA-Z0-9_-]{95}$/
-              ),
-              sharedSecret: expect.stringMatching(/^[a-zA-Z0-9-_]{43}$/),
-              assetCode: incomingPayment.receivedAmount.assetCode,
-              assetScale: incomingPayment.receivedAmount.assetScale
-            }
-      }),
+      getBody: (incomingPayment, list) => {
```

Comment:
Could this instead be
```suggestion
      getBody: (incomingPayment, list) => incomingPayment.toOpenPaymentsType(paymentPointer)
```
with some possible special assignment for `ilpStreamConnection`?

Reply:
In terms of the connection, we would be able to do something like:
```ts
getBody: (incomingPayment, list) => ({
        ...incomingPayment.toOpenPaymentsType(paymentPointer),
        ilpStreamConnection: list
          ? `${config.openPaymentsUrl}/connections/${incomingPayment.connectionId}`
          : {
              id: `${config.openPaymentsUrl}/connections/${incomingPayment.connectionId}`,
              ilpAddress: expect.stringMatching(
                /^test\.rafiki\.[a-zA-Z0-9_-]{95}$/
              ),
              sharedSecret: expect.stringMatching(/^[a-zA-Z0-9-_]{43}$/),
              assetCode: incomingPayment.receivedAmount.assetCode,
              assetScale: incomingPayment.receivedAmount.assetScale
            }
      }),
```

Since `incomingPayment.toOpenPaymentsType` expect a `Connection`, we wouldn't be able to use `expect.stringMatching(...)`, but would have to create a `Connection` instance with proper `credentials` etc

Reply:
i.e. I'm not sure if that's a much better solution